crawlers/mooncrawl/mooncrawl/eth_nft_explorer.py

```diff
@@ -0,0 +1,195 @@
+from dataclasses import dataclass, asdict
+from collections import defaultdict
+from typing import Dict, List, Optional
+from web3 import Web3
+import web3
+from web3.types import FilterParams
+from web3._utils.events import get_event_data
+
+
+w3 = Web3(Web3.HTTPProvider("http://127.0.0.1:18375"))
+
+# First abi is for old NFT's like crypto kitties
+# The erc721 standart requieres that Transfer event is indexed for all arguments
+# That is how we get distinguished from erc20 transfer events
+erc721_transfer_event_abis = [
+    {
+        "anonymous": False,
+        "inputs": [
+            {"indexed": False, "name": "from", "type": "address"},
+            {"indexed": False, "name": "to", "type": "address"},
+            {"indexed": False, "name": "tokenId", "type": "uint256"},
+        ],
+        "name": "Transfer",
+        "type": "event",
+    },
+    {
+        "anonymous": False,
+        "inputs": [
+            {"indexed": True, "name": "from", "type": "address"},
+            {"indexed": True, "name": "to", "type": "address"},
+            {"indexed": True, "name": "tokenId", "type": "uint256"},
+        ],
+        "name": "Transfer",
+        "type": "event",
+    },
+]
+
+erc721_functions_abi = [
+    {
+        "inputs": [{"internalType": "address", "name": "owner", "type": "address"}],
+        "name": "balanceOf",
+        "outputs": [{"internalType": "uint256", "name": "", "type": "uint256"}],
+        "payable": False,
+        "stateMutability": "view",
+        "type": "function",
+        "constant": True,
+    },
+    {
+        "inputs": [],
+        "name": "name",
+        "outputs": [{"internalType": "string", "name": "", "type": "string"}],
+        "stateMutability": "view",
+        "type": "function",
+        "constant": True,
+    },
+    {
+        "inputs": [{"internalType": "uint256", "name": "tokenId", "type": "uint256"}],
+        "name": "ownerOf",
+        "outputs": [{"internalType": "address", "name": "", "type": "address"}],
+        "payable": False,
+        "stateMutability": "view",
+        "type": "function",
+        "constant": True,
+    },
+    {
+        "inputs": [],
+        "name": "symbol",
+        "outputs": [{"internalType": "string", "name": "", "type": "string"}],
+        "stateMutability": "view",
+        "type": "function",
+        "constant": True,
+    },
+    {
+        "inputs": [],
+        "name": "totalSupply",
+        "outputs": [{"internalType": "uint256", "name": "", "type": "uint256"}],
+        "stateMutability": "view",
+        "type": "function",
+        "constant": True,
+    },
+]
+
+
+@dataclass
+class NFT_contract:
+    address: str
+    name: str
+    symbol: str
+    total_supply: str
+
+
+def get_erc721_contract_info(address: str) -> NFT_contract:
```

Comment:
Let's remove for now since we're not using it in the crawler.

Reply:
Ah, this might be useful when analyzing top NFT contracts. Disregard my previous comment.